src/devices/nm-device.c

```diff
@@ -4194,8 +4203,15 @@ device_init_static_sriov_num_vfs (NMDevice *self)
 		                                          NULL);
 		num_vfs = _nm_utils_ascii_str_to_int64 (value, 10, 0, G_MAXINT32, -1);
 		if (num_vfs >= 0) {
-			nm_platform_link_set_sriov_params (nm_device_get_platform (self),
-			                                   priv->ifindex, num_vfs, NM_TERNARY_DEFAULT);
+			nm_clear_g_cancellable (&priv->sriov_cancellable);
+			priv->sriov_cancellable = g_cancellable_new ();
+			nm_platform_link_set_sriov_params_async (nm_device_get_platform (self),
+			                                         priv->ifindex,
```

Comment:
you cannot just "cancel && create-cancelleable && set-async"

If an async operation is going on, you need to cancel it, wait for the callback to return, and then invoke the async operation. Otherwise, there is a race.